cmd/kwil-admin/nodecfg/toml.go

```diff
@@ -257,4 +273,34 @@ max_tx_bytes = {{ .ChainCfg.Mempool.MaxTxBytes }}
 
 # Size of the cache (used to filter transactions we saw earlier) in transactions
 cache_size = {{ .ChainCfg.Mempool.CacheSize }}
+
+#######################################################
+###         State Sync Configuration Options        ###
+#######################################################
+[chain.statesync]
+# State sync rapidly bootstraps a new node by discovering, fetching, and restoring a state machine
+# snapshot from peers instead of fetching and replaying historical blocks. Requires some peers in
+# the network to take and serve state machine snapshots. State sync is not attempted if the node
+# has any local state (LastBlockHeight > 0). The node will have a truncated block history,
+# starting from the height of the snapshot.
+enable = {{ .ChainCfg.StateSync.Enable }}
+
+# Trusted snapshot providers (comma-separated chain RPC servers) are the source-of-truth for the snapshot integrity.
+# Snapshots are accepted for statesync only after verifying it with these trusted snapshot providers.
+# These are also used for light client verification of the synced state machine and
+# retrieval of state data for node bootstrapping.
+# Light client verification needs a trusted height and block hash at the configured block height obtained from a
+# trusted source, and a period during which validators can be trusted.
+# Atleast 2 trusted rpc servers are required for enabling state sync.
+rpc_servers = "{{ .ChainCfg.StateSync.RPCServers }}"
+trust_height = {{ .ChainCfg.StateSync.TrustHeight }}
+trust_hash = "{{ .ChainCfg.StateSync.TrustHash }}"
+trust_period = "{{ .ChainCfg.StateSync.TrustPeriod }}"
```

Comment:
These trust settings are missing from cmd/kwild/config/default_config.toml and kwil-db/cmd/kwild/config/test_data/config.toml

Reply:
Actually, how does one set these?  I kinda though everything from `rpc_servers` was trusted.

Reply:
https://docs.cometbft.com/v0.34/core/light-client#where-to-obtain-trusted-height--hash

Reply:
I see this in cometbft:

```go
// TrustOptions are the trust parameters needed when a new light client
// connects to the network or when an existing light client that has been
// offline for longer than the trusting period connects to the network.
//
// The expectation is the user will get this information from a trusted source
// like a validator, a friend, or a secure website. A more user friendly
// solution with trust tradeoffs is that we establish an https based protocol
// with a default end point that populates this information. Also an on-chain
// registry of roots-of-trust (e.g. on the Cosmos Hub) seems likely in the
// future.
type TrustOptions struct {
	// tp: trusting period.
	//
	// Should be significantly less than the unbonding period (e.g. unbonding
	// period = 3 weeks, trusting period = 2 weeks).
	//
	// More specifically, trusting period + time needed to check headers + time
	// needed to report and punish misbehavior should be less than the unbonding
	// period.
	Period time.Duration

	// Header's Height and Hash must both be provided to force the trusting of a
	// particular header.
	Height int64
	Hash   []byte
}
```

The part "The expectation is the user will get this information from a trusted source like a validator, a friend, or a secure website. A more user friendly  solution with trust tradeoffs is that we establish an https based protocol with a default end point that populates this information" what what I thought rpc_servers did.

Reply:
<img width="681" alt="Screenshot 2024-04-26 at 4 40 08 PM" src="https://github.com/kwilteam/kwil-db/assets/45089429/8b728203-990a-4bce-9da8-5db837aafc36">

Umm, but cometbft rejects statesync if these configs are not provided, what the new node does is, it uses these trust options and queries the rpc servers and verifies these options before starting statesync.

Reply:
I noticed that too. But in practice, how is a kwild operator going to set these?  IT sounds to me that they'd literally be using the same RPC provider to get these `trust_` settings as the host(s) then use for the `rpc_servers` setting.

Reply:
yup.

Reply:
https://medium.com/tendermint/everything-you-need-to-know-about-the-tendermint-light-client-f80d03856f98

Essentially, they do this sequential verification mentioned in this doc, starting from this trust height.

But yeah, they would query the known rpc endpoints, get a block header and populate this information.